### PR TITLE
Neutral naming of insurer

### PR DESCRIPTION
**Pull request by MarkusTiede**
Neutralize naming

### CODE REVIEW

**Comment by MarkusTiede** [comment]
@frank-baier - do you happen to have the src-Files for the interaction process chart? If so could you please add it to the /docs section? I'd like to neutralize the naming inside.

File: docs/openapi.json [hunk 1]
```diff
@@ -13,25 +13,46 @@
...
-      "get" : {
-        "tags" : [ "version" ],
-        "summary" : "callable way of retrieving current version",
-        "description" : "callable way (compared to analyzing the URL of the endpoint) of retrieving current version. As well good for testing purposes as operation is get only",
-        "operationId" : "version",
+    "/contracts/v1/documents" : {
+      "post" : {
+        "tags" : [ "documents" ],
+        "summary" : "Upload a document for a contract.",
+        "description" : "Please note that this operation needs to be called per document for a contract to be uploaded. The response contains a handle to the document. This handle should be provided with the contract to be uploaded via uploadContract",
```
Comment (dweber019) [inline docs/openapi.json]: I would use the perma linking to point to the call where the handle can be used.
E.g. https://stackoverflow.com/questions/52703804/how-to-link-to-another-endpoint-in-swagger

File: docs/openapi.json [hunk 2]
```diff
@@ -13,25 +13,46 @@
...
+                "$ref" : "#/components/schemas/Document"
+              }
+            }
+          },
+          "required" : true
+        },
         "responses" : {
           "default" : {
-            "description" : "version identifier",
+            "description" : "A handle to the document for providing with a new contract to be uploaded via POST contracts",
```
Comment (dweber019) [inline docs/openapi.json]: Use perma linking to point to the contract.
Consider rephrase the sentence to something like:
A handle for the uploaded document which can be used in...

File: docs/openapi.json [hunk 3]
```diff
@@ -13,25 +13,46 @@
...
+              "application/json" : {
+                "schema" : {
+                  "$ref" : "#/components/schemas/ErrorResponse"
                 }
               }
             }
           },
           "503" : {
-            "description" : "technical issue on our side, please retry later",
+            "description" : "Technical issue on our side, please retry later",
```
Comment (dweber019) [inline docs/openapi.json]: I think it makes sense to be neutral in naming => "our"
Would rephrase to "Technical issue on server side"

File: docs/openapi.json [hunk 4]
```diff
@@ -94,40 +115,19 @@
...
-              "application/json" : {
-                "schema" : {
-                  "$ref" : "#/components/schemas/FileHandle"
-                }
-              }
-            }
-          },
-          "400" : {
-            "description" : "Invalid Document is provided. See Error details for more information about validation issues",
+            "description" : "version identifier",
```
Comment (dweber019) [inline docs/openapi.json]: Start uppercase

File: docs/openapi.json [hunk 5]
```diff
@@ -94,40 +115,19 @@
...
-                "$ref" : "#/components/schemas/Document"
-              }
-            }
-          },
-          "required" : true
-        },
+    "/contracts/v1/version" : {
+      "get" : {
+        "tags" : [ "version" ],
+        "summary" : "Callable way of retrieving current version",
```
Comment (dweber019) [inline docs/openapi.json]: Would rephrase to be more specific.
Which version? API?

File: docs/openapi.json [hunk 6]
```diff
@@ -94,40 +115,19 @@
...
+        "description" : "Callable way (compared to analyzing the URL of the endpoint) of retrieving current version. As well good for testing purposes as operation is get only",
```
Comment (dweber019) [inline docs/openapi.json]: Would rephrase to "Endpoint to retrieve the current API version. Can be compared to the URI version. Can be used for testing purposes."

File: docs/openapi.json [hunk 7]
```diff
@@ -148,13 +148,13 @@
...
-        "required" : [ "value" ],
+      "FileHandle" : {
+        "required" : [ "handle" ],
         "type" : "object",
         "properties" : {
-          "value" : {
+          "handle" : {
             "type" : "string",
-            "description" : "version identifier of the current API definition"
+            "description" : "Reference to an uploaded file. This can be e.g. a PDF or a picture (e.g. JPG) of a contract or a purchase receipt."
```
Comment (dweber019) [inline docs/openapi.json]: Rephrase to "Reference to an uploaded document. This can be a PDF or a picture (e.g. JPG) of a contract or a purchase receipt."

File: docs/openapi.json [hunk 8]
```diff
@@ -181,14 +181,48 @@
           }
         }
       },
+      "Document" : {
+        "required" : [ "contractId", "data", "documentType", "fileName", "mediaType" ],
+        "type" : "object",
+        "properties" : {
+          "data" : {
+            "type" : "array",
+            "description" : "the document itself",
```
Comment (dweber019) [inline docs/openapi.json]: - Start uppercase
- Describe the exact format (Base64)

File: docs/openapi.json [hunk 9]
```diff
@@ -181,14 +181,48 @@
...
+            "items" : {
+              "type" : "string",
+              "format" : "byte"
+            }
+          },
+          "contractId" : {
+            "maxLength" : 20,
+            "minLength" : 0,
+            "type" : "string",
+            "description" : "the contract this document belongs to"
```
Comment (dweber019) [inline docs/openapi.json]: Start uppercase

File: docs/openapi.json [hunk 10]
```diff
@@ -181,14 +181,48 @@
...
+          },
+          "mediaType" : {
+            "type" : "string",
+            "description" : "accepted media types for documents represented by data byte[]",
```
Comment (dweber019) [inline docs/openapi.json]: Start uppercase

File: docs/openapi.json [hunk 11]
```diff
@@ -515,37 +539,13 @@
...
-          "fileName" : {
-            "maxLength" : 256,
-            "minLength" : 0,
-            "type" : "string"
-          },
-          "documentType" : {
+          "value" : {
             "type" : "string",
-            "enum" : [ "contract", "receipt" ]
+            "description" : "version identifier of the current API definition"
```
Comment (dweber019) [inline docs/openapi.json]: Start uppercase

File: docs/swagger.json [hunk 12]
```diff
@@ -87,13 +87,13 @@
...
           "503": {
-            "description": "technical issue on our side, please retry later",
+            "description": "Technical issue on our side, please retry later",
             "schema": {
               "$ref": "#/definitions/ErrorResponse"
             }
           },
           "default": {
-            "description": "a handle to the document for providing with a new contract to be uploaded via POST contracts",
+            "description": "A handle to the document for providing with a new contract to be uploaded via POST contracts",
```
Comment (dweber019) [inline docs/swagger.json]: Use perma link

File: docs/openapi.json (hunk 2 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done

File: docs/openapi.json (hunk 3 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done

File: docs/openapi.json (hunk 4 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done

File: docs/openapi.json (hunk 5 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done

File: docs/openapi.json (hunk 9 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done

File: docs/openapi.json (hunk 10 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done

File: docs/openapi.json (hunk 11 above)
Comment (MarkusTiede) [reply docs/openapi.json]: ok - done
